lam4doc/entrypoints/api/handlers.py

```diff
@@ -39,3 +44,63 @@ def generate_lam_report() -> tuple:
     except Exception as e:
         logger.exception(str(e))
         raise InternalServerError(str(e))  # 500
+
+
+def upload_rdfs(body: dict, lam_properties_document: FileStorage = None, lam_classes_document: FileStorage = None,
+                celex_classes_document: FileStorage = None) -> tuple:
+    """
+    API method for uploading 3 RDF files.
+    :return: nothing
+    """
+    logger.debug('Entering upload_rdfs')
+
+    dataset_name = body.get("dataset_name")
+    if not dataset_name:
+        logger.error("The name of the dataset is required.")
+        return 'Dataset name is required', 400
+
+    try:
+        logger.debug(body)
+        sparql_adapter = FusekiSPARQLAdapter(LAM_FUSEKI_LOCATION + ":" + str(LAM_FUSEKI_PORT) + "/", requests)
+
+        if lam_properties_document is None and lam_classes_document is None and celex_classes_document is None:
+            logger.error("No file was specified. Returning 400.")
+            return "Please supply at least one file: lam_properties_document, lam_classes_document, " \
+                   "celex_classes_document", 400
+
+        with tempfile.TemporaryDirectory() as temp_folder:
+            if lam_properties_document:
```

Comment:
this code block seems to be repeated 3 times; a new function shall be implemented and used accordingly.

Reply:
This functionality is encapsulated in the logic of the method and is not needed anywhere else. I would not extract it just for the fact that it repeats, because that would expose it to the outside of the method. Maybe extract it when somebody else needs it ?